devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1333,14 +1353,30 @@ void HumanStateProvider::impl::computeSecondaryCalibrationRotationsForChain(cons
         if (!(wearableStorage.modelToWearable_LinkName.find(linkToCalibrateName) == wearableStorage.modelToWearable_LinkName.end())) {
             // discarding previous calibration
             eraseSecondaryCalibration(linkToCalibrateName);
-            // computing new calibration
-            iDynTree::Rotation linkRotationZero = kinDynComputations->getWorldTransform(linkToCalibrateName).getRotation();
-            iDynTree::Rotation secondaryCalibrationRotation = linkTransformMatricesRaw.at(linkToCalibrateName).getRotation().inverse() * linkRotationZero;
+
+            iDynTree::Transform linkTransformZero = kinDynComputations->getWorldTransform(linkToCalibrateName);
+
+            // computing new calibration for orientation
+            iDynTree::Rotation secondaryCalibrationRotation = linkTransformMatricesRaw.at(linkToCalibrateName).getRotation().inverse() * linkTransformZero.getRotation();
+
+            // compute secondary calibration for position
+            iDynTree::Position secondaryCalibrationPosition = linkTransformZero.getPosition() - linkTransformMatricesRaw.at(linkToCalibrateName).getPosition();
```

Comment:
Just curious, why this would be `p1 - p2` and not be `p1 - (R1' * p2)` ? They're both already in the same coordinate frames?

Reply:
Yes, I think you are right. Anyway, actually are not using secondary calibration for the position of the links (see https://github.com/lrapetti/human-dynamics-estimation/blob/add-base-calibration/devices/HumanStateProvider/HumanStateProvider.cpp#L1556) since we are just considering the link orientation for the inverse kinematics, and the base offset is applied together with the [secondaryCalibrationWorld](https://github.com/lrapetti/human-dynamics-estimation/blob/add-base-calibration/devices/HumanStateProvider/HumanStateProvider.cpp#L1559) which contains both position and orientation offset of the world.

I will probably proceed directly removing the `secondaryCalibrationPosition`

Reply:
Removed in https://github.com/robotology/human-dynamics-estimation/pull/237/commits/941ab91c08b2637a19795c7be83761849a1401f2